### ISSUE TITLE
Support netcat like behaviour for az network bastion tunnel

### ISSUE THREAD

**Issue by kirat-singh** [Azure/azure-cli-extensions#5807]
### Related command
az network bastion tunnel

### Extension name (the extension in question)
ssh-extension

### Description of issue (in as much detail as possible)
It would be great if the CLI would have an option to behave like netcat, read from stdin and forward to the tunnel, and read from the tunnel and write to stdout.

This would avoid hardcoding a local port which prevents any useful scripting.

It would also allow it to be used as ProxyCommand within ssh to seamlessly ssh into an azure virtual machine.

As it stands, this requires running a persistent tunnel in one window, and then ssh in another window.  And it's impossible to script it in a unix environment.
-----

**Comment by yonzhan** [comment]
route to CXP team

**Comment by kurarrr** [comment]
+1

**Comment by msapitree** [comment]
I am pretty surprised so few voted/asked for this. Anyway, any feedback on the topic, please? @aznetsuppgithub

**Comment by 0x416e746f6e** [comment]
whoa, more than a year.  sad, so very sad

**Comment by mpcusack-sequelae** [comment]
@0x416e746f6e  you can wrap it with something like:
```
#!/usr/bin/env bash

SUBSCRIPTION="$1"
BASTION="$2"
RESOURCE_GROUP="$3"
VM_ID="$4"
TUNNEL_PORT="${5:-50022}"

echo "Opening a proxy tunnel to $VM_ID on port $TUNNEL_PORT and exposing it to stdin/stdout for ssh." > /dev/stderr

az network bastion tunnel \
    --subscription "$SUBSCRIPTION" \
    --name "$BASTION" \
    --resource-group "$RESOURCE_GROUP" \
    --target-resource-id "$VM_ID" \
    --resource-port 22 \
    --port "$TUNNEL_PORT" &
tunnel_pid="$!"

function kill_tunnel() {
    kill $tunnel_pid
    echo "Tunnel Killed $?" > /dev/stderr
}
trap kill_tunnel INT

echo "Tunnel pid: $tunnel_pid" > /dev/stderr
sleep 4
socat "TCP:localhost:$TUNNEL_PORT" -
echo "END" > /dev/stderr
```

**Comment by azzid** [comment]
Worked after enabling CRB-repo for CentOS.

**Comment by 0x416e746f6e** [comment]
@mpcusack-sequelae yeah, I thought of a script like that, but it would be cumbersome to port it across different user machines.  e.g. in non-azure cases it's easy to just manage `ssh_config` files in a git repo that users would just refer in their respective `ssh_config` local files with `Include` directive.  but with the shell-script used as a proxy-command, they would have to clone such repo in a very specific location, which is ugly and error-prone.

all the while the irony is that this bastion command is deployed with a python-plugin sitting in `~/.azure/cliextensions/bastion/azext_bastion`.  by the looks of it it would take probably half-day to implement stdin/stdout redirecting wrapper around it, which I would be happy to do, but it's not even open-source!  🤷‍♂️

**Comment by 0x416e746f6e** [comment]
t.w.i.m.c. I ended up with this atrocity in my ssh_config:

```ssh_config
Host name-of-the-instance-in-azure
    HostName      name-of-the-instance-in-azure
    ProxyCommand  bash -c "port=$(( 4096 + $RANDOM %% 1024 )) ; pgid=$( ps -o pgid= -p $$ ) ; az network bastion tunnel --resource-group REDACTED --name REDACTED --resource-port %p --port \$port --target-resource-id /subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Compute/virtualMachines/%h & trap \"kill -- -\$pgid\" INT QUIT TERM EXIT ; sleep 2 ; nc 127.0.0.1 \$port"
    User          azureuser
```

**Comment by tpo** [comment]
@0x416e746f6e 's hack ...

```
Host name-of-the-instance-in-azure
     HostName      name-of-the-instance-in-azure
     ProxyCommand  bash -c "port=$(( 4096 + $RANDOM %% 1024 )) ; pgid=$( ps -o pgid= -p $$ ) ; az network bastion tunnel --resource-group REDACTED --name REDACTED --resource-port %p --port \$port --target-resource-id /subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Compute/virtualMachines/%h & trap \"kill -- -\$pgid\" INT QUIT TERM EXIT ; sleep 2 ; nc 127.0.0.1 \$port"
     User          azureuser
```

... is the most impressive SW engineering I must've seen in a while and qualifies for an ACM Turing Award IMO. It'd be awesome to combine this with https://stackoverflow.com/a/78125240 (some bash script to find a free port). I've tried but I got completely tangled up in "quo\'te e$(cap\"ing he" and gave up. The $RANDOM hit and miss heuristic does the job for now...

Thanks a log @0x416e746f6e !!!
